### ISSUE TITLE
twisted.internet.task.LoopingCall does not document a public API for checking to see if it is currently running.

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2725]
|[<img alt="dreid's avatar" src="https://avatars.githubusercontent.com/u/48695?s=50" width="50" height="50">](https://github.com/dreid)| @dreid reported|
|-|-|
|Trac ID|trac#2725|
|Type|defect|
|Created|2007-06-26 23:19:57Z|
|Branch|https://github.com/twisted/twisted/tree/loopingcall-2725|

LoopingCall defines a 'running' attribute but does not document it at as public API.  If it is a public API it should be documented, if it is not a public API then it should be renamed with an _ to indicate it is private.

Attachments:

* [LoopingCall.running.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/fc3/fc339cfad6bffa85d804db823d8f387cdea57cb3/4d17bebaa64c4aabae67b8c88fc25fc4a9bbc61a.patch) (557 bytes) - added by jerith on 2011-12-29 20:19:06Z - 
* [LoopingCall.running.2.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/fc3/fc339cfad6bffa85d804db823d8f387cdea57cb3/cb10eaa8d3d85ce1fa88118d0fee79028b30e585.patch) (1235 bytes) - added by jerith on 2011-12-31 16:02:33Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__2725 2725
type__defect defect
reporter__dreid dreid
priority__low low
milestone__ 
branch__branches_loopingcall_2725 branches/loopingcall-2725
branch_author__jerith jerith
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1182899997000000 1182899997000000
changetime__1326662385000000 1326662385000000
version__None None
owner__itamar itamar
cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun set owner to @dreid|
|-|-|

I disagree with the assertion in the description of this ticket.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_3"></a>|@thijstriemstra commented|
|-|-|

Replying to [exarkun](#note_1):
> I disagree with the assertion in the description of this ticket.

What's the status of this ticket then now?

**Comment by twisted-trac** [comment]
|[<img alt="jerith's avatar" src="https://avatars.githubusercontent.com/u/43731?s=50" width="50" height="50">](https://github.com/jerith)<a name="note_4"></a>|@jerith commented|
|-|-|

I've seen this come up a few times in #twisted, so I've attached a two-line documentation patch.

**Comment by twisted-trac** [comment]
|[<img alt="MostAwesomeDude's avatar" src="https://avatars.githubusercontent.com/u/118035?s=50" width="50" height="50">](https://github.com/MostAwesomeDude)<a name="note_5"></a>|@MostAwesomeDude commented|
|-|-|

LGTM. I don't have commit powers, though, so somebody else needs to be okay with it.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun set owner to @jerith|
|-|-|

Thanks.  I'd like to see a couple more things done with this before the patch is applied:

  1. `C{}` is for containing things that could maybe be Python expressions.  So you probably want `C{bool}` not `C{boolean}`.
  1. I think it's a little obvious that a flag named _running_ would tell you whether the `LoopingCall` is running or not.  What would make this documentation addition really useful is if it explained what _running_ actually means.  Some useful things to say might be that `running` is `True` between calls to `start` and `stop`, and that it will be `False` by the time the `Deferred` returned by `start` fires.

Thanks again!

**Comment by twisted-trac** [comment]
|[<img alt="jerith's avatar" src="https://avatars.githubusercontent.com/u/43731?s=50" width="50" height="50">](https://github.com/jerith)<a name="note_7"></a>|@jerith removed owner|
|-|-|

Updated patch.

1. I copied C{boolean} from somewhere else without thinking. Fixed.
2. Good point. I've rewritten it to be a little more explanatory, while hopefully retaining clarity.

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_8"></a>|@itamarst commented|
|-|-|

(In [33438]) Branching to 'loopingcall-2725'

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_9"></a>|@itamarst set owner to @itamarst<br>@itamarst set status to `assigned`|
|-|-|

Needed news file, which I added. I will merge in a second.

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_10"></a>|@itamarst set status to `closed`|
|-|-|

(In [33440]) Merge loopingcall-2725.

Author: jerith
Review: exarkun, itamar
Fixes: [#2725](https://github.com/twisted/twisted/issues/2725)

Document the running attribute of LoopingCall.
